plot_equality_data.R

```diff
@@ -0,0 +1,280 @@
+            which(compare_table$ID == paste(PRTPnum, etanum, "NA", "WLD", sep="_")), column
+          ]])
+        )
+      }
+    }
+  }
+}
+if (type_str == "poor_pref_10dollars"){
+  compare_results$value = compare_results$value * dollar_val_2020  
+}
+if (grepl("nice", dmg_f)){
+  nice_data <- read.csv("./data/nice_imports/interpolated_scc_scenarios_v4_2025.csv")
+  nice_data$ssp = substr(nice_data$Scenario, 4, 4)
+  nice_data$rcp = sapply(nice_data$Scenario, function(x) strsplit(x, "-")[[1]][2])
+  nice_data$rcp[nice_data$rcp=="Baseline"] = "85"
+  nice_data_poor = nice_data[nice_data$whose=="poorest", ]
+  len_nice = nrow(nice_data_poor)
+  nice_results_table <- data.table(
+    ssp=integer(), rcp=numeric(), eta=numeric(), PRTP=numeric(), damages=rep("NICE", len_nice), 
+    indicator=rep("mean", len_nice), value=numeric())
+  nice_results_table$ssp = nice_data_poor$ssp
+  nice_results_table$rcp = nice_data_poor$rcp
+  nice_results_table$PRTP = as.integer(nice_data_poor$prtp * 100)
+  nice_results_table$eta = nice_data_poor$eta
+  nice_results_table$value = nice_data_poor$scc
+  if (type_str == "poor_pref_10dollars") {
+    # The post-abatement cost, post-damages income of the poorest group in the NICE model in 2025.
+    poorest_nice = 0.34 * 1e3 
+    nice_results_table$value = poorest_nice * (nice_results_table$value / 10)**(-1/nice_results_table$eta)
+    results_table = rbind(results_table, nice_results_table)
+  } else {
+    results_table = rbind(results_table, nice_results_table)
+  }
+}
+
+
+if(type_str == "eri_eq_statscc_2020d"){
+  plot_labs = labs(
+    title="World SCC relative to the income of an Eritrean",
+    fill="Inequality aversion"
+  ) 
+  plot_ylab = ylab("SCC (2020 USD)") 
+} else if(type_str == "poor_pref_10dollars"){
+  plot_labs = labs(
+    title="Income at which direct donation is preferable to $10/ton CO2",
+    fill="Inequality aversion"
+  )
+  plot_ylab = ylab("Yearly income (2020 USD)") 
+}
+
+set.seed(10)
+explain_eta = function(et){paste0("RRA: ", et)}
+explain_prtp = function(pr){paste0("PRTP: ", pr)}
+label_axes = labeller(
+  eta=explain_eta, PRTP=explain_prtp
+)
+results_table$SSP <- factor(results_table$ssp)
+plot = ggplot(results_table, aes(x=rcp, y=value, color=SSP))+geom_point(
+  alpha=0.9, size=2
+) + plot_labs + plot_ylab + xlab("RCP pathway") + facet_grid(PRTP+eta~damages , labeller = label_axes
+)
+if (type_str != "eri_eq_statscc_2020d"){
+  plot = plot + geom_hline(yintercept=1.9*365)
+  plot = plot + geom_hline(yintercept=500, color="red")
+} else {
+  plot = plot + geom_hline(yintercept=10)
+}
+plot = plot + scale_y_continuous(trans = "log2")
+
+plot
+
+subDir = "plots"
+if (!dir.exists(file.path(subDir))){
+  dir.create(file.path(subDir))
+}
+
+savefig = paste0(type_str, version_string, ".png")
+ggsave(path="plots", filename=savefig)
+
+if (type_str == "eri_eq_statscc_2020d"){
+  combined_data = results_table
+  combined_data$ratio = compare_results$value / results_table$value
```

Comment:
Error, because compare_results has 540 entries and results_table has 570. So NICE values should also be added to compare_results table